app/models/facility.rb

```diff
@@ -9,11 +9,25 @@
 #  created_at        :datetime         not null
 #  updated_at        :datetime         not null
 #  image_description :string(255)
+#  image_file        :string(255)
 #
 
 class Facility < ActiveRecord::Base
   has_many :court_facilities
-  attr_accessible :image, :name, :image_description
+  attr_accessible :image, :name, :image_description, :image_file
+  validates :image_file, presence: true
+  validate :image_dimensions
 
   default_scope { order('LOWER(name)') } # ignore case when sorting
+
+  mount_uploader :image_file, FacilityImageUploader
+
+  private
+
+  def image_dimensions
+    return if self.image_file.blank?
+    if self.image_file.width != 50 || self.image_file.height != 50
```

Comment:
If the image dimensions have to be 50x50px then maybe this `if` statement should check both width and heigh values rather than just one them.

How about to change it to `if self.image_file.width != 50 && self.image_file.height != 50` ?